c/meterpreter/source/extensions/priv/service.c

```diff
@@ -172,3 +172,47 @@ DWORD service_destroy( char * cpName )
 
 	return dwResult;
 }
+
+/*
+ * Query service state.
+ */
+DWORD query_service_status( char* cpName, DWORD* dwState )
+{
+	DWORD dwResult  = ERROR_SUCCESS;
+	HANDLE hManager = NULL;
+	HANDLE hService = NULL;
+	DWORD dwBytes;
+	SERVICE_STATUS_PROCESS procInfo;
+
+	do
+	{
+		if( !cpName )
+			BREAK_WITH_ERROR( "[SERVICE] query_service_status. cpName is NULL", ERROR_INVALID_HANDLE );
+
+		hManager = OpenSCManagerA( NULL, NULL, SC_MANAGER_CONNECT);
+		if( !hManager )
+			BREAK_ON_ERROR( "[SERVICE] query_service_status. OpenSCManagerA failed" );
+
+		hService = OpenServiceA( hManager, cpName, SERVICE_QUERY_STATUS);
+		if( !hService )
+			BREAK_ON_ERROR( "[SERVICE] query_service_status. OpenServiceA failed" );
```

Comment:
May want to update this to print out the `cpName` parameter so we know what service we are trying to open?